### DISCUSSION TITLE
Airflow sentry integration - Allow use of custom transports

### DISCUSSION THREAD

**Discussion by luispedroemg** [apache/airflow#30257] [Ideas]
Is there any reason why the sentry integration does not allow setting a custom transport?

My use-case is that I require airflow sentry events to be sent to different DSN's based on the DAG that the error or error log occurred in.

I was trying to follow the suggestions in this discussion: https://github.com/getsentry/sentry-python/discussions/1484
But I then noticed that setting the transport option in the sentry integration for airflow is unsupported.

Is there any reason why we cannot set a "multiplexing transport" as described in the discussion I linked?

I imagine it would work similarly to setting the before_send function (set a dotted path to the transport implementation we want to use).

**Comment by potiuk** [comment]
PRs to support more options are always welcome. I think you are asking wrong question. Is there any reason people like you who get a free software are not contributing (or paying someone if they are not skilled enough) to contribute for the features they would like to use.

I think you should answer the question first.

**Reply by potiuk** [reply]
Both contributions, and paying someone to do it are most welcome!

**Reply by luispedroemg** [reply]
Don't get me wrong... I am planning on opening a PR ... just making sure that I was not going to make a PR for a feature that definitely wouldn't get accepted because of some decision not to support the feature....
Just assuming that my question was a "rant" is also very discouraging for everyone else wanting to post questions/open discussions...
Notice I asked if there was a reason why it was not supported... I never asked for anyone else to implement it....

**Reply by luispedroemg** [reply]
So... @potiuk I ask again... Is there a reason why it was not supported or is it cool if I try to implement support for it?

**Reply by potiuk** [reply]
@luispedroemg Answering again: I have no idea. And again - you are asking wrong question. The right question is "can I improve the code and stand behind it?",

Airlfow has > 2400 contributors - often people like you who contribute things where they see the need. Maintainers review the code and if the justification is sound and they are convinced the change is good, they merge it.

Airflow has also almost 80 integrations and more than 5000 provider operators. Other than looking at the code/commit history and issues (which you can do as well) there is no way anyone can answer authoritatively about all of them - especially in so insignificant part of airflow.

If you think it is cool and you convince maintainer who will decide to look at your code (and maybe others who can chime in), and your code will be good, it will get merged.

**Reply by luispedroemg** [reply]
@potiuk Thanks for clarifying. I'll work on trying to get this feature through
